src/SSEConnection.js

```diff
@@ -0,0 +1,110 @@
+import AbstractConnection from "./AbstractConnection";
+import ConnectionEvent from "./ConnectionEvent";
+import { List } from "immutable";
+
+/**
+ * Server Sent Event Connections
+ */
+export default class SSEConnection extends AbstractConnection {
+  /**
+   * Initializes an Instance of SSEConnection
+   * @constructor
+   * @param {string} url – URL to connect to
+   * @param {object} [options]
+   * @param {boolean} [options.withCredentials=true] – used method
+   */
+  constructor(url, options = {}) {
+    super(url);
+
+    Object.defineProperty(this, "options", {
+      value: { withCredentials: options.withCredentials || false }
+    });
+  }
+
+  /**
+   * callback for server sent events
+   *
+   * @callback eventCallback
+   * @param {object} [event]
+   * @param {string} [event.data]
+   */
+
+  /**
+   * Adds an event listener for the type
+   * @param {string} type
+   * @param {eventCallback} callback
+   */
+  addEventListener(type, callback) {
+    const addListenerToSource = () =>
+      this.source.addEventListener(type, callback);
+
+    if (this.state === SSEConnection.OPEN) {
+      addListenerToSource();
+    } else {
+      // not yet started, so defer the initialization
+      this.on(ConnectionEvent.OPEN, addListenerToSource);
+    }
+  }
+
+  /**
+   * Removes an event listener for the type
+   * @param {string} type
+   * @param {eventCallback} callback
+   */
+  removeEventListener(type, callback) {
+    const removeListenerFromSource = () =>
+      this.source.removeEventListener(type, callback);
+
+    if (this.state === SSEConnection.OPEN) {
+      removeListenerFromSource();
+    } else {
+      // not yet started, so defer the removal
+      this.on(ConnectionEvent.OPEN, removeListenerFromSource);
```

Comment:
the reason for deferring in not clear to me

Reply:
Because the source is not there yet. We could also remove the deferred adding, but that would mean we need to have some kind of queue for the "to be added"-listeners instead of a callback, which seemed harder than this.

Reply:
fair enough